library/alloc/src/boxed.rs

```diff
@@ -283,9 +283,7 @@ impl<T> Box<T> {
     #[must_use]
     #[inline(always)]
     pub fn pin(x: T) -> Pin<Box<T>> {
-        (#[rustc_box]
-        Box::new(x))
-        .into()
+        Box::new(x).into()
```

Comment:
In an unoptimized build, `T` already needs to be on the stack, so this change shouldn't inhibit any in-place heap initialization.